html/includes/api_functions.inc.php

```diff
@@ -1305,33 +1108,29 @@ function list_bills()
     global $config;
     $app = \Slim\Slim::getInstance();
     $router = $app->router()->getCurrentRoute()->getParams();
-    $status = 'ok';
-    $err_msg = '';
-    $message = '';
-    $code = 200;
+
     $bills = array();
     $bill_id = mres($router['bill_id']);
     $bill_ref = mres($_GET['ref']);
     $bill_custid = mres($_GET['custid']);
+    $param = array();
+    $sql = '1';
     if (!empty($bill_custid)) {
-        $sql   = '`bill_custid` = ?';
-        $param = array($bill_custid);
+        $sql    .= '`bill_custid` = ?';
+        $param[] = $bill_custid;
     } elseif (!empty($bill_ref)) {
-        $sql   = '`bill_ref` = ?';
-        $param = array($bill_ref);
+        $sql    .= '`bill_ref` = ?';
+        $param[] = $bill_ref;
     } elseif (is_numeric($bill_id)) {
-        $sql   = '`bills`.`bill_id` = ?';
-        $param = array($bill_id);
-    } else {
-        $sql   = '';
-        $param = array();
+        $sql    .= '`bills`.`bill_id` = ?';
+        $param[] = $bill_id;
     }
-
-    if (count($param) >= 1) {
-        $sql = "WHERE $sql";
+    if (!is_admin() && !is_read()) {
+        $sql    .= ' AND `bills`.`bill_id` IN (SELECT `bill_id` FROM `bills_perms` WHERE `user_id` = ?)';
+        $param[] = $_SESSION['user_id'];
     }
 
-    foreach (dbFetchRows("SELECT `bills`.*,COUNT(port_id) AS `ports_total` FROM `bills` LEFT JOIN `bill_ports` ON `bill_ports`.`bill_id`=`bills`.`bill_id` $sql GROUP BY `bill_name`,`bill_ref` ORDER BY `bill_name`", $param) as $bill) {
+    foreach (dbFetchRows("SELECT `bills`.*,COUNT(port_id) AS `ports_total` FROM `bills` LEFT JOIN `bill_ports` ON `bill_ports`.`bill_id`=`bills`.`bill_id` WHERE $sql GROUP BY `bill_name`,`bill_ref` ORDER BY `bill_name`", $param) as $bill) {
```

Comment:
This old query is bad. Use:

`foreach (dbFetchRows("SELECT `bills`.* FROM `bills` WHERE $sql ORDER BY `bill_name`", $param) as $bill) {`

Reply:
I did see this, along with the query inside the loop to populate ports, but as this had been approved as an addition before I left it.